internal/docker/docker.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 SAP SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	util "github.com/sapcc/go-makefile-maker/internal"
+	"github.com/sapcc/go-makefile-maker/internal/core"
+)
+
+func mustI(_ int, err error) {
+	util.Must(err)
+}
+
+func RenderConfig(cfg core.Configuration) error {
+	var goBuildflags, packages, user, entrypoint string
+
+	template, err := os.ReadFile("Dockerfile.template")
+	util.Must(err)
+
```

Comment:
As discussed in person, please add the `Dockerfile.template` to this repo. We will have Renovate Bot update our Go versions and Alpine versions once in this repo, and then these updates get cascaded down into the application repos via the less-ops-more-dev automation. For how to embed files into Go code at build time, check out <https://pkg.go.dev/embed>.